url-sampler/src/main/java/io/opentelemetry/contrib/samplers/StringAttributeSampler.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.contrib.samplers;
+
+import static java.util.stream.Collectors.toMap;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingDecision;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This sampler drops spans which have one of attribute values matching given pattern.
+ */
+public class StringAttributeSampler implements Sampler {
+  private final Map<AttributeKey<String>, StringMatcher> matchers;
+  private final SpanKind kind;
+  private final Sampler delegate;
+
+  public StringAttributeSampler(Map<AttributeKey<String>, ? extends Collection<String>> patterns, SpanKind kind, Sampler delegate) {
```

Comment:
It is more generic, but is it really more complex? I think "setting a delegate Sampler for a pattern" is about as simple as it gets conceptually, it's almost like just populating a key/value set at that point. And while healthcheck is probably the #1 ask, it is common for systems to have APIs with vastly different QPS, so we can expect the new #1 ask to immediately want to be able to tweak their sampling rate :) I suspect it is worth working on this expectation, it doesn't seem to me to make things overly complex.

Reply:
Wait. I realised that your suggestion is totally different from my original idea. I wanted to have "deny list": if attribute matches one of the patterns, then drop, otherwise delegate to the "main" sampler. E.g. my main sampler for the whole application is ParentBased, but I want to drop "/healthcheck". Deny list.

Your idea is more like "router": match some patterns to samplers. Which means that I have to provides `AlwaysOff` sampler for "/healthcheck" pattern and I have to remember to provide "catchAll" pattern to direct to `ParentBased` sampler. Router.

We probably may want both samplers, one simpler (including simpler to think about), one more powerful.